### PR TITLE
Add slack notifications

### PR DESCRIPTION
**Pull request by zzak**

### CODE REVIEW

File: api/api.go [hunk 1]
```diff
@@ -18,8 +18,6 @@ import (
 	yaml "gopkg.in/yaml.v2"
 )
 
-// GQLErrorsCollection is a slice of errors returned by the GraphQL server.
-// Each error is made up of a GQLResponseError type.
```
Comment (yundt) [inline api/api.go]: Just making sure, as it's not connected to the main PR's topic: was this removal intentional?

File: .circleci/config.yml [hunk 2]
```diff
@@ -3,6 +3,7 @@ version: 2.1
 orbs:
   codecov: circleci/codecov-clojure@0.0.4
   shellcheck: circleci/shellcheck@1.2.0
+  slack-orb: circleci/slack@dev:volatile
```
Comment (yundt) [inline .circleci/config.yml]: If we want to rely on those slack notifications in our workflow, maybe it would be safer to pin the version instead of using `volatile`?

Also, a nit: we don't use `-orb` for other aliases; we could change it to just `slack`, and then rename the `slack` job below to something like `slack-notify-on-failure`?

Comment (zzak) [reply .circleci/config.yml]: We're waiting on a release of the slack orb, that's backed up due to the master branch failing before we can change from `volatile`. (Actually the PR I wrote was never deployed because of this).

Regarding the naming, good idea; I will change this to what you suggested.

File: api/api.go (hunk 1 above)
Comment (zzak) [reply api/api.go]: Correct, this was removed in 43a2846 so we could see an actual notification in slack
